File: tools/bench/answers/python_haiku/s1/14_rect_lib.py
class Rectangle:
    def __init__(self, width, height):
        self.width = width
        self.height = height

def area(rect):
    return rect.width * rect.height

def perimeter(rect):
    return 2 * (rect.width + rect.height)

def scale(rect, factor):
    return Rectangle(rect.width * factor, rect.height * factor)

rect = Rectangle(3.0, 4.0)
print(area(rect))
print(perimeter(rect))
scaled = scale(rect, 2.0)
print(area(scaled))

File: tools/bench/answers/go_haiku/s3/14_rect_lib.go
package main

import (
	"fmt"
)

type Rectangle struct {
	width  float64
	height float64
}

func (r Rectangle) area() float64 {
	return r.width * r.height
}

func (r Rectangle) perimeter() float64 {
	return 2 * (r.width + r.height)
}

func (r Rectangle) scale(factor float64) Rectangle {
	return Rectangle{r.width * factor, r.height * factor}
}

func main() {
	rect := Rectangle{3.0, 4.0}
	fmt.Println(rect.area())
	fmt.Println(rect.perimeter())
	scaled := rect.scale(2.0)
	fmt.Println(scaled.area())
}
